src/components/dropdown/index.coffee

```diff
@@ -26,19 +26,24 @@ positionDropdown = ({alignments, selection, dropdown, useScroll}, {matchWidth})
   zIndex = parentZIndex(selection.node(), true)
 
   # calculate the position of the dropdown
-  {x, y} = calculateDropdownPosition(
-    alignments,
+  {x, y, direction} = calculateDropdownPosition(
+    dropdown.attr('data-direction') or align,
     { x: rect.left, y: rect.top, width: rect.width, height: rect.height },
     { width: dropdownRect.width, height: dropdownRect.height },
     { width: window.innerWidth, height: window.innerHeight },
     ddMaxHeight,
-    scrollbarSize()
+    scrollbarSize(),
   )
 
+  yPos = 'top'
   if not parentFixed
     x += window.scrollX || window.pageXOffset
     y += window.scrollY || window.pageYOffset
 
+    if direction is 'up'
+      yPos = 'bottom'
+      y = document.body.clientHeight - y - dropdownRect.height
```

Comment:
Taken from the old hexagon code. I'm not sure when this logic was removed (or why) but it's mostly required for browsers that don't support the `MutationObserver` (as otherwise the increase/decrease in size is applied from the top of the dropdown, causing an overlap with the selection)